subprojects/gradle/build-metrics/src/main/kotlin/com/avito/android/plugin/build_metrics/CollectTeamcityMetricsTask.kt

```diff
@@ -21,12 +23,15 @@ abstract class CollectTeamcityMetricsTask @Inject constructor(
 
     @TaskAction
     fun action() {
-        @Suppress("UnstableApiUsage")
-        workerExecutor.noIsolation().submit(CollectTeamcityMetricsWorkerAction::class.java) { parameters ->
-            parameters.getBuildId().set(buildId)
-            parameters.getLogger().set(project.ciLogger)
-            parameters.getGraphiteConfig().set(project.graphiteConfig)
-            parameters.getTeamcityCredentials().set(project.teamcityCredentials)
+        workerExecutor.inMemoryWork {
```

Comment:
The fix

Reply:
i don't get the purpose of this `inMemoryWork` extension.
Why not use standard gradle worker api's and make things serializable? Isn't that possible?

No offense for your current workaround, just think how things should be

Reply:
If I'm not mistaken, `inMemoryWork` was created to reduce boilerplate for `InstrumentationTestsAction.Params`.
I'll try to use a standard way. I don't know the impact on performance\resources. We create a new logger for each task.

Reply:
`SentryDestination` uses `Project.logger`. It's non-serializable and we can't store it inside `CILogger`.
It means that we can't pass `Project.logger` to an action.
I see nothing for logging in injectable services: https://docs.gradle.org/6.0/userguide/custom_gradle_types.html#service_injection

For now, I see one simple workaround - use `System.err` directly.

Reply:
https://gradle-community.slack.com/archives/CA745PZHN/p1608129333272700

Reply:
Reverted this change and started using `System.err` inside the destination.

Reply:
> SentryDestination uses Project.logger

i don't see how :(

Reply:
Here, in lambda inside [ElasticFactory](https://github.com/avito-tech/avito-android/commit/8f1963f7fb56be2ae562405b04fcf6674f9cce0c#diff-521faccf554aa9b4bb1a749be1a9138d5b711ea1575bbc12e17bdbd47c629093R37)